tools/stonith_admin.c

```diff
@@ -631,7 +674,8 @@ main(int argc, char **argv)
         case 'M':
             if (agent == NULL) {
                 printf("Please specify an agent to query using -a,--agent [value]\n");
-                return CRM_EX_USAGE;
+                exit_code = CRM_EX_USAGE;
+                goto done;
             } else {
                 char *buffer = NULL;
 
```

Comment:
Options  --list-installed  and --metadata work well for lha fence agents.

The only thing is --metadata  doesn't give any stderror if being supplied with a non-existing fence agent. Only if running with -V, we could see:
```
(stonith_get_namespace)         error: Unknown fence agent: aaa
```
And with -VVVV, we also get:
```
(main)  info: Command returned: Invalid argument (-22)
```

Probably put a stderr if st->cmds->metadata() fails in the following?

Reply:
Good point -- I think it was like that already, but it's a good opportunity to fix it.

Reply:
And this also looks good to me.Thanks.